jit-service/src/main/java/org/dcsa/jit/service/TimestampDefinitionService.java

```diff
@@ -61,6 +62,69 @@ public void markOperationsEventAsTimestamp(OperationsEvent operationsEvent) {
     opsEventTimestampDefinitionRepository.save(ops);
   }
 
+  public PortCallPhaseTypeCode findPhaseTypeCodeFromOperationsEventForJit1_0(
+      OperationsEvent operationsEvent) {
+
+    List<TimestampDefinition> jit1_0 =
+        timestampDefinitionRepository
+            .findByEventClassifierCodeAndOperationsEventTypeCodeAndProvidedInStandardAndPortCallServiceTypeCodeAndFacilityTypeCode(
+                operationsEvent.getEventClassifierCode(),
+                operationsEvent.getOperationsEventTypeCode(),
+                "jit1_0",
+                operationsEvent.getPortCallServiceTypeCode(),
+                operationsEvent.getFacilityTypeCode());
+
+    String errorMessage =
+        "EventClassifierCode: "
+            + operationsEvent.getEventClassifierCode()
+            + ", OperationsEventTypeCode: "
+            + operationsEvent.getOperationsEventTypeCode()
+            + ", PortCallServiceTypeCode: "
+            + operationsEvent.getPortCallServiceTypeCode()
+            + ", FacilityTypeCode: "
+            + operationsEvent.getFacilityTypeCode();
+
+    // Not found in Jit 1.0, trying Jit 1.1 instead
+    if (jit1_0.isEmpty()) {
+
+      List<TimestampDefinition> jit1_1 =
+          timestampDefinitionRepository
+              .findByEventClassifierCodeAndOperationsEventTypeCodeAndProvidedInStandardAndPortCallServiceTypeCodeAndFacilityTypeCode(
+                  operationsEvent.getEventClassifierCode(),
+                  operationsEvent.getOperationsEventTypeCode(),
+                  "jit1_1",
+                  operationsEvent.getPortCallServiceTypeCode(),
+                  operationsEvent.getFacilityTypeCode());
+
+      if (jit1_1.size() > 1) {
+        // If all PortCallPhaseTypeCodes are the same we can just return the first
+        if (jit1_1.stream()
+            .allMatch(
+                x -> jit1_1.get(0).getPortCallPhaseTypeCode() == x.getPortCallPhaseTypeCode())) {
+          return jit1_1.get(0).getPortCallPhaseTypeCode();
+        }
+        throw ConcreteRequestErrorMessageException.internalServerError(
+            "More than one JIT 1.1 timestamp definitions found according to the given fields: "
+                + errorMessage);
+      }
+
+      if (jit1_1.isEmpty()) {
+        throw ConcreteRequestErrorMessageException.internalServerError(
+            "No JIT 1.1 timestamp definitions found according to the given fields: "
+                + errorMessage);
+      }
+      return jit1_1.stream().findFirst().get().getPortCallPhaseTypeCode();
```

Comment:
I think we should be strict and reject JIT 1.1+ timestamps without port call phase type code (unless the timestamp explicitly permit it to be null - none do right now as I recall, but some will soon).